### DISCUSSION TITLE
Running tests locally results in garbage output

### DISCUSSION THREAD

**Discussion by abyss7** [jelmer/dulwich#939] [Q&A] [answered]
Running like `python -m subunit.run dulwich.tests.test_suite` and get something like - don't even know if tests pass:

2�+@n[ub2�O#@[dulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_clone_from_dulwich_empty��+@h[ub2�T@Udulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwichyȅJ�+@h[ub3�  @Udulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich�5��+@}[ub3��X@jdulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich_issue_88_alternative���L�+@}[ub3���@jdulwich.tests.compa�8@gdulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich_issue_88_standard�lCM�+@z[ub4�]�H@gdulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich_issue_88_standardߦ{~�+@n[ub4�e�x@[dulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich_no_op)g��+@n[ub5�Sި@[dulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich_no_op &�z�+@�[ub5�Z

**Comment by jelmer** [comment, marked as answer]
On Thu, Aug 16, 2018 at 04:45:36AM -0700, abyss7 wrote:
> Running like `python -m subunit.run dulwich.tests.test_suite` and get something like - don't even know if tests pass:
> 
> 2�+@n[ub2�O#@[dulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_clone_from_dulwich_empty��+@h[ub2�T@Udulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwichyȅJ�+@h[ub3�  @Udulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich�5��+@}[ub3��X@jdulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich_issue_88_alternative���L�+@}[ub3���@jdulwich.tests.compa�8@gdulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich_issue_88_standard�lCM�+@z[ub4�]�H@gdulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich_issue_88_standardߦ{~�+@n[ub4�e�x@[dulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich_no_op)g��+@n[ub5�Sި@[dulwich.tests.compat.test_server.GitServerSideBand64kTestCase.test_fetch_from_dulwich_no_op &�z�+@�[ub5�Z

You're running the command to generate subunit
(https://github.com/testing-cabal/subunit) formatted test results.

To run the testsuite with regular output, use a standard python test
runner. E.g.:

python -m unittest dulwich.tests.test_suite

-- 
Jelmer Vernooĳ <jelmer@jelmer.uk>
PGP Key: https://www.jelmer.uk/D729A457.asc
